### PR TITLE
Included CSISnapshotController API Type

### PR DESCRIPTION
**Pull request by huffmanca**
This includes the CSISnapshot API, needed for the CSISnapshotOperator being introduced in 4.4.

Note that this also updates the `Dockerfile.build` from F29 to F30, as this is required to run the `make generate-with-container` command. Otherwise there will be errors on cmd/godoc due to ee.ExitCode being undefined.

/cc @gnufied @jsafrane for review.

### CODE REVIEW

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  version: v1
+  versions:
+  - name: v1
+    served: true
+    storage: true
+  validation:
+    openAPIV3Schema:
+      description: KubeAPIServer provides information to configure an operator to
+        manage kube-apiserver.
+      type: object
```
Comment (gnufied) [inline operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: description needs fixing.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,140 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: csisnapshots.operator.openshift.io
+spec:
+  group: operator.openshift.io
+  names:
+    kind: CSISnapshot
```
Comment (gnufied) [inline operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: Is it too late to call this `CSISnapshotController` ?

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: CSISnapshotSpec is the specification of the desired behavior of
+            the CSISnapshot.
```
Comment (gnufied) [inline operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: It is desired behaviour of `CSI Snapshot operator` not the snapshot itself.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            unsupportedConfigOverrides:
+              description: 'unsupportedConfigOverrides holds a sparse config that
+                will override any previously set options.  It only needs to be the
+                fields to override it will end up overlaying in the following order:
+                1. hardcoded defaults 2. observedConfig 3. unsupportedConfigOverrides'
+              type: object
+              nullable: true
+              x-kubernetes-preserve-unknown-fields: true
+        status:
+          description: CSISnapshotStatus defines the observed status of the CSISnapshot.
```
Comment (gnufied) [inline operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: all these fields define behaviour of snapshot controller/operator. not of snapshot objects.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml (hunk 1 above)
Comment (huffmanca) [reply operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: This has been corrected for the current CSISnapshot object. If the Kind changes, then I'll update this again.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml (hunk 2 above)
Comment (huffmanca) [reply operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: I'm not opposed, but would want insight from @jsafrane as well. If needed I can change this.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml (hunk 3 above)
Comment (huffmanca) [reply operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: I've updated the description to indicate that this is for the operator and not the snapshot.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml (hunk 4 above)
Comment (huffmanca) [reply operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: I've updated the description to indicate that this is for the operator and not the snapshot.

File: operator/v1/types_csi.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package v1
```
Comment (gnufied) [inline operator/v1/types_csi.go]: please rename this file to something like `types_csi_snapshot.go`

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml (hunk 2 above)
Comment (jsafrane) [reply operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: CSISnapshotController sounds good to me (folowing KubeControllerManager CRD name).

File: operator/v1/types_csi.go (hunk 5 above)
Comment (huffmanca) [reply operator/v1/types_csi.go]: This has been renamed and updated.

File: operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml (hunk 2 above)
Comment (huffmanca) [reply operator/v1/0000_80_csi_snapshot_operator_01_crd.yaml]: This has been renamed. The references in both `types_csi_controller.go` and this file have been updated to match.

**Comment by jsafrane** [comment]
CRD file should be probably `operator/v1/0000_80_csi_snapshot_controller_operator_01_crd.yaml`

**Comment by huffmanca** [comment]
> CRD file should be probably `operator/v1/0000_80_csi_snapshot_controller_operator_01_crd.yaml`

This has been corrected.

**Comment by jsafrane** [comment]
lgtm

**Pull request merged**